From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Phil Rodgers <philip.rodgers@glasgow.ac.uk>
Date: Wed, 21 Oct 2026 10:42:58 +0000
Subject: [PATCH 3/3] bin: use mimalloc on native targets only

mimalloc builds C code. On wasm targets the binary uses the default
allocator.
---
 Cargo.toml      | 4 +++-
 src/bin/main.rs | 2 ++
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/Cargo.toml b/Cargo.toml
index 0ccf496..2827211 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -46,7 +46,6 @@ env_logger = "0.11.11"
 git-version = "0.3.9"
 itertools = "0.15.0"
 log = "0.4"
-mimalloc = "=0.1.48"
 thiserror = "2.0.20"
 
 [dependencies.veripb-checker]
@@ -110,6 +109,9 @@ panic = "abort"
 [profile.bench]
 debug = true
 
+[target.'cfg(not(target_family = "wasm"))'.dependencies]
+mimalloc = "=0.1.48"
+
 [features]
 default = ["native-compression"]
 native-compression = ["veripb-parser/native-compression"]
diff --git a/src/bin/main.rs b/src/bin/main.rs
index 7bc5a2b..155d6ac 100644
--- a/src/bin/main.rs
+++ b/src/bin/main.rs
@@ -3,12 +3,14 @@ use clap::Parser;
 use colored::Colorize;
 use git_version::git_version;
 use log::Level;
+#[cfg(not(target_family = "wasm"))]
 use mimalloc::MiMalloc;
 use std::io::Write;
 
 use veripb::{args::Args, run_checker};
 
 /// Use [mimalloc](https://github.com/microsoft/mimalloc) to improve allocation performance.
+#[cfg(not(target_family = "wasm"))]
 #[global_allocator]
 static GLOBAL: MiMalloc = MiMalloc;
 
